### DISCUSSION TITLE
Libuv worker pool

### DISCUSSION THREAD

**Discussion by pioardi** [poolifier/poolifier#232] [Ideas]
@jerome-benoit I was thinking that we could ( optionally based on a constructor option ) delegate a part of the tasks to the libuv threads, we could need to write a C++ addon for that  but you said you are a C programmer right ? https://nodejs.org/en/docs/guides/dont-block-the-event-loop/#what-code-runs-on-the-worker-pool

Just an idea, to verify if it is really possible, who knows

CC: @Shinigami92

**Comment by jerome-benoit** [comment]
If node.js have an API to write bindings to C, C++, etc..., that's doable. 
And if there's already an npm package implementing such bindings for libuv, that's even better.

**Reply by Shinigami92** [reply]
![image](https://user-images.githubusercontent.com/7195563/108863392-74ad1380-75f1-11eb-9649-e22e6dc84ba7.png)

**Reply by pioardi** [reply]
I know @Shinigami92 , we can evaluate an external binding eventually if is too hard to write.
But we will look into this in the future :)

**Reply by pioardi** [reply]
We first need  to document our selection strategies and cover them fully and release the 2.0.0 version right ? :P

**Reply by jerome-benoit** [reply]
I guess that if we want to use libuv worker, that will need to change. Because even if we wrote it from stratch, it will not be part of poolifier but an external package consumed by it.
